### PR TITLE
use config file from home, create if it doesn't exist

### PR DESCRIPTION
**Pull request by andy5995**

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -1,14 +1,74 @@
...
-        return "./telluricdeckayrc".to_string();
+        let basename = "telluricdeckayrc".to_owned();
+        let cwd = "./".to_owned();
+        let default = cwd + &basename;
+        if std::path::Path::new(&default).exists() {
+            return default;
+        }
+        let config_home = match env::var("XDG_CONFIG_HOME") {
+            Ok(val) => val,
+            Err(_e) => get_homedir().unwrap() + "/.config",
```
Comment (Jammyjamjamman) [inline src/config.rs:53]: Mental note: if getting XDG_CONFIG_HOME and homedir fails, the program crashes.

File: src/config.rs [hunk 2]
```diff
@@ -1,14 +1,74 @@
...
+        if !std::path::Path::new(&config_file).exists() {
+            write_config(&config_file, c_data).expect("Error writing config file");
+        }
+
+        return config_file;
     }
     opt_cfg.unwrap()
 }
 
+// I think the comment below is outdated. Is there any good reason that
```
Comment (Jammyjamjamman) [inline src/config.rs]: I suspect that comment was for something else. I think it's fine to delete

- [ ] delete lines 69-72.

**Review by Jammyjamjamman — CHANGES_REQUESTED** [review]
Looks fine to me. Is the code actually used yet? I tried running telluricdeckay but (as far as I can tell), no config is generated.

There's some comments L69-72 I think can go.

File: src/config.rs (hunk 1 above)
Comment (andy5995) [reply src/config.rs:53]: > Looks fine to me. Is the code actually used yet? I tried running telluricdeckay but (as far as I can tell), no config is generated.

No, not yet.

I added a TODO comment for handing an error from get_homedir(). I think we'll have a better idea of how to handle the error as we complete the code base more.

Comment (Jammyjamjamman) [reply src/config.rs:53]: tbh panicking may be ok. Might be sensible to fail if there's no homedir

Comment (andy5995) [reply src/config.rs:53]: My thoughts exactly. But I think better to exit from error-handling than panic, right? (or is that C thinking?)

**Pull request merged**